src/udp_packets.rs

```diff
@@ -0,0 +1,277 @@
+pub mod udp_payload {
+    // const MAX_PAYLOAD_IPV4: usize = 65507; // 65,535 - 20 (IP header) - 8 (UDP header)
+    // const MAX_PAYLOAD_IPV6: usize = 65527; // 65,535 - 40 (IPv6 header) - 8 (UDP header)
+
+    /// From RFC 768
+    /// We send blank UDP packets, unless the port is determined to be special.
+    ///
+    /// ```text
+    /// HEADER Handled by the OS
+    /// 0                   1                   2                   3
+    /// 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+    /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    /// |          Source Port          |       Destination Port        |
+    /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    /// |            Length             |           Checksum            |
+    /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    /// We append the data, based on the port the the header
+    /// DATA
+    /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    /// |                    Data (based on port)                       |
+    /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    /// ```
+    ///
+    /// # Fields
+    ///
+    /// - **Source Port**: The source port number.
+    /// - **Destination Port**: The destination port number.
+    /// - **Length**: The length of the UDP header and data.
+    /// - **Checksum**: The checksum for error-checking.
+    /// - **Data**: The payload data, which can vary based on the port.
+    pub fn cust_payload(dst_prt: u16) -> Vec<u8> {
+        match dst_prt {
+            // 53 => craft_dns_query_packet(),
+            // 67 => craft_dhcpc_packet(),
+            // 123 => craft_ntp_packet(),
+            // 161 => craft_snmp_packet(),
+            _ => vec![],
+        }
+    }
+
+    pub fn craft_snmptrap_packet() -> Vec<u16> {
+        let version = vec![0x30, 0x82]; // SNMP version
+        let community_string = vec![0x04, 0x06, 0x70, 0x75, 0x62, 0x6c, 0x69, 0x63]; // Community string "public"
+        let pdu_type = vec![0xa4, 0x82]; // PDU type for SNMP Trap
+        let enterprise_oid = vec![0x06, 0x09, 0x2b, 0x06, 0x01, 0x02, 0x01, 0x01, 0x01, 0x00]; // Enterprise OID
+
+        // Combine the components into a single SNMP Trap packet
+        [version, community_string, pdu_type, enterprise_oid].concat()
+    }
+
+    /// Crafts an SNMP packet based on the structure from RFC 1592.
+    ///
+    /// RFC 1592 defines the structure of an SNMP packet as follows
```

Comment:
question: worth linking to the actual RFC?

As above, is it worth referencing the official doc?